include/alp/utils/parser/MatrixFileIterator.hpp

```diff
@@ -0,0 +1,377 @@
+
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * @author A. N. Yzelman
+ * @date 25th of May, 2017
+ */
+
+#ifndef _H_MATRIXFILEITERATOR
+#define _H_MATRIXFILEITERATOR
+
+#include <cstddef> //std::ptrdiff_t
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <limits>
+#include <sstream>
+#include <stdexcept> //std::runtime_error
+
+#include <stdlib.h> //posix_memalign
+
+#include <utility> //std::pair
+
+#include <assert.h>
+
+#include <alp/config.hpp>
+#include <alp/iomode.hpp>
+//#include <alp/spmd.hpp>
+//#include <alp/utils/config.hpp>
+//#include <alp/utils/hpparser.h>
+
+#include "MatrixFileProperties.hpp"
+
+namespace alp {
+	namespace utils {
+		namespace internal {
+
+			template< typename T, bool data_reflect = false,  typename S = size_t >
+			class MatrixFileIterator {
+
+			private:
+				/** The output type of the base iterator. */
+				typedef T OutputType;
+
+				/** The underlying MatrixReader. */
+				MatrixFileProperties & properties;
```

Comment:
No space between `&` and variable name. Please check the same in the remaining code.